lib/kitchen/driver/openstack.rb

```diff
@@ -308,7 +308,9 @@ def get_ip(server)
         end
 
         # make sure we have the latest info
-        server.reload
+        info 'Waiting for network information to be available...'
+        w = server.wait_for { reload && !addresses.empty? }
```

Comment:
This adds an implicit failure if `addresses` is empty. I wanted to get this out, so I didn't add any logic for rescuing from the timeout exception or anything else.

Note: the default timeout is 5 minutes, but we can pass in a different one if we choose.